fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java

```diff
@@ -0,0 +1,227 @@
+                            IcebergFileStats.toMap(idToTypeMapping, dataFile.upperBounds()),
+                            dataFile.nullValueCounts(),
+                            dataFile.columnSizes());
+                } else {
+                    icebergFileStats.incrementFileCount();
+                    icebergFileStats.incrementRecordCount(dataFile.recordCount());
+                    icebergFileStats.incrementSize(dataFile.fileSizeInBytes());
+                    updateSummaryMin(icebergFileStats, partitionFields, IcebergFileStats.toMap(idToTypeMapping,
+                            dataFile.lowerBounds()), dataFile.nullValueCounts(), dataFile.recordCount());
+                    updateSummaryMax(icebergFileStats, partitionFields, IcebergFileStats.toMap(idToTypeMapping,
+                            dataFile.upperBounds()), dataFile.nullValueCounts(), dataFile.recordCount());
+                    icebergFileStats.updateNullCount(dataFile.nullValueCounts());
+                    updateColumnSizes(icebergFileStats, dataFile.columnSizes());
+                }
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        if (icebergFileStats == null) {
+            return Statistics.builder().build();
+        }
+
+        Statistics.Builder statisticsBuilder = Statistics.builder();
+        double recordCount = icebergFileStats.getRecordCount();
+        for (Map.Entry<Integer, String> idColumn : idToColumnNames.entrySet()) {
+            List<ColumnRefOperator> columnList = colRefToColumnMetaMap.keySet().stream().filter(key -> {
+                LOG.debug("test key name " + key.getName());
+                LOG.debug("test id column name " + idColumn.getValue());
+                return key.getName().equals(idColumn.getValue());
+            }).collect(Collectors.toList());
+            if (columnList == null || columnList.size() != 1) {
+                LOG.debug("This column is not required " + (columnList == null ? "null" : columnList.size()));
+                continue;
+            }
+
+            int fieldId = idColumn.getKey();
+            ColumnStatistic.Builder columnBuilder = ColumnStatistic.builder();
+            Long nullCount = icebergFileStats.getNullCounts().get(fieldId);
+            if (nullCount != null) {
+                columnBuilder.setNullsFraction(nullCount / recordCount);
+            }
+            if (icebergFileStats.getColumnSizes() != null) {
+                Long columnSize = icebergFileStats.getColumnSizes().get(fieldId);
+                if (columnSize != null) {
+                    columnBuilder.setAverageRowSize(columnSize / recordCount);
+                }
+            }
+            Object min = icebergFileStats.getMinValues().get(fieldId);
+            Object max = icebergFileStats.getMaxValues().get(fieldId);
+            if (min instanceof Number && max instanceof Number) {
```

Comment:
for date and datetime, it will also return numeric value
```
switch (type.typeId()) {
      case BOOLEAN:
        return tmp.get() != 0x00;
      case INTEGER:
      case DATE:
        return tmp.getInt();
      case LONG:
      case TIME:
      case TIMESTAMP:
        if (tmp.remaining() < 8) {
          // type was later promoted to long
          return (long) tmp.getInt();
        }
        return tmp.getLong();
```